pkg/cli/alpha/internal/update/update.go

```diff
@@ -0,0 +1,372 @@
+	}
+	defer func() {
+		if err = response.Body.Close(); err != nil {
+			log.Errorf("failed to close the connection: %v", err)
+		}
+	}()
+
+	if response.StatusCode != http.StatusOK {
+		return "", fmt.Errorf("failed to download the binary: HTTP %d", response.StatusCode)
+	}
+
+	_, err = io.Copy(file, response.Body)
+	if err != nil {
+		return "", fmt.Errorf("failed to write the binary content to file: %w", err)
+	}
+
+	if err := os.Chmod(binaryPath, 0o755); err != nil {
+		return "", fmt.Errorf("failed to make binary executable: %w", err)
+	}
+	log.Infof("release binary version %s successfully downloaded to %s", version, binaryPath)
+	return tempDir, nil
+}
+
+// cleanupBranch removes all files and folders in the current directory
+// except for the .git directory and the PROJECT file.
+// This is necessary to ensure the ancestor branch starts with a clean slate
+// TODO: Analise if this command is still needed in the future.
+// It is required because the alpha generate command in versions prior to v4.7.0 do not properly
+// handle the removal of files in the ancestor branch.
+func cleanupBranch() error {
+	log.Info("Cleaning all files and folders except .git and PROJECT")
+	cmd := exec.Command("sh", "-c", "find . -mindepth 1 -maxdepth 1 ! -name '.git' ! -name 'PROJECT' -exec rm -rf {} +")
+
+	if err := cmd.Run(); err != nil {
+		return fmt.Errorf("failed to clean up files: %w", err)
+	}
+	log.Info("Successfully cleanup files in ancestor branch")
+	return nil
+}
+
+// runMakeTargets is a helper function to run make with the targets necessary
+// to ensure all the necessary components are generated, formatted and linted.
+func runMakeTargets() {
+	targets := []string{"manifests", "generate", "fmt", "vet", "lint-fix"}
+	for _, target := range targets {
+		err := util.RunCmd(fmt.Sprintf("Running make %s", target), "make", target)
+		if err != nil {
+			log.Warnf("make %s failed: %v", target, err)
+		}
+	}
+}
+
+// runAlphaGenerate executes the old Kubebuilder version's 'alpha generate' command
+// to create clean scaffolding in the ancestor branch. This uses the downloaded
+// binary with the original PROJECT file to recreate the project's initial state.
+func runAlphaGenerate(tempDir, version string) error {
+	log.Infof("Generating project with version %s", version)
+	// Temporarily modify PATH to use the downloaded Kubebuilder binary
+	tempBinaryPath := tempDir + "/kubebuilder"
+	originalPath := os.Getenv("PATH")
+	tempEnvPath := tempDir + ":" + originalPath
+
+	if err := os.Setenv("PATH", tempEnvPath); err != nil {
+		return fmt.Errorf("failed to set temporary PATH: %w", err)
+	}
+
+	defer func() {
+		if err := os.Setenv("PATH", originalPath); err != nil {
+			log.Errorf("failed to restore original PATH: %v", err)
+		}
+	}()
+
+	// TODO: we need improve the implementation from utils to allow us
+	// to pass the path of the binary and use it to run the alpha generate command.
+	cmd := exec.Command(tempBinaryPath, "alpha", "generate")
```

Comment:
@vitorfloriano we have in the utils the command and a code centralized to do things like
We need to modify the utils implementation or determine how to utilise it to allow us to pass the path and avoid duplications. It can be a follow-up